core/registry.ts

```diff
@@ -184,16 +168,32 @@ export function register<T>(
  * @param registryItem A class or object that we are checking for the required
  *     properties.
  */
-function validate(type: string, registryItem: Function | AnyDuringMigration) {
+function validateItem(
+  type: string,
+  registryItem: Function | AnyDuringMigration,
+) {
   switch (type) {
     case String(Type.FIELD):
       if (typeof registryItem.fromJson !== 'function') {
-        throw Error('Type "' + type + '" must have a fromJson function');
+        throw Error(`Type "${type}" must have a fromJson function`);
       }
       break;
   }
 }
 
+/**
+ * Checks that the name hasn't been registered using a different case.
+ *
+ * @param type The type of the plugin. (e.g. Field, Renderer)
+ * @param name The plugin's name. (Ex. field_angle, geras)
+ */
+function validateCase(type: string, name: string) {
```

Comment:
I think this is now a breaking change in the other direction, we previously allowed you to ignore case (except for serializers, where the name you pass in is actually in the serialized output) so throwing an error now will break anyone who was relying on that. 

There might still be some cleanup that's possible here but I don't think we should change the behavior in either direction, in other words, we live with the somewhat weird state that most registries are case-insensitive except for the serializer. And the cleanup doesn't need to block the allow duplicate registration of values since that's required for v11 so there's some urgency there, unlike the cleanup.

Reply:
The current PR will break two cases:
1. Where someone is using our partial case-sensitivity to store different values for keys 'foo' and 'Foo'.
2. Where someone is using our partial case-insensitivity to get away with inconsistently switching between 'foo' and 'Foo'.
I'd argue that both of these behaviours are bugs.

It would be one thing if we could just leave this mess alone, but we need to build on top of it for V11.  This problem is going to contiune to compound, and we'll have a registry that behaves in random ways and is impossible to document.  I think if we don't straighten the registry out now, it's just going to get worse and worse.

Reply:
I don't think it's fair to call those behaviors bugs because we explicitly built support for them (i.e. the parameter you can pass to allow for case insensitivity). In hindsight, we may have made a different decision around this if we went back in time. But now, if we want to change it, we need to do a proper cleanup which would mean weighing the value of this breaking change against the burden of communicating it to users and forcing them to do work to change their code if they're affected by it. Part of that proper cleanup would be doing a real breaking change, communicating it effectively, and not having it be mixed with the feature of allowing duplicate registration.

I hear you that we're building on top of mess, but the actual change we need for v11 doesn't have anything to do with case sensitivity and is really a very small change buried in the much larger case sensitivity refactoring part of this PR, so I don't think that's a compelling enough reason to mix the two issues.

If you want to file an issue for the case insensitivity cleanup, we can triage it separately during our normal bug triage process. I would prefer you do just the duplicate registration of values in this PR, and wait to see how prioritization works out for the rest of the cleanup vs everything else we already have planned for the v11 release